### PR TITLE
Added test to ensure users may not sign up with phone number already in use

### PR DESCRIPTION
**Pull request by sbryzak**
Added e2e test for https://github.com/codeready-toolchain/registration-service/pull/168

Fixes https://issues.redhat.com/browse/CRT-883

This PR also includes an increase to the timeout of client requests to the host API, to allow a greater threshold when running in CRC.

Signed-off-by: Shane Bryzak <shane@localhost.localdomain>

### CODE REVIEW

File: test/e2e/registration_service_test.go [hunk 1]
```diff
@@ -495,6 +495,35 @@ func (s *registrationServiceTestSuite) TestPhoneVerification() {
...
+	// Wait for the UserSignup to be created
+	otherUserSignup, err := s.hostAwait.WaitForUserSignup(otherIdentity.ID.String(),
+		wait.UntilUserSignupHasConditions(VerificationRequired()...),
+		wait.UntilUserSignupHasStateLabel(v1alpha1.UserSignupStateLabelValueNotReady))
+	require.NoError(s.T(), err)
+	otherEmailAnnotation := userSignup.Annotations[v1alpha1.UserSignupUserEmailAnnotationKey]
+	assert.Equal(s.T(), otherEmailValue, otherEmailAnnotation)
+
+	// Initiate the verification process using the same phone number as previously
+	invokeEndpoint(s.T(), "PUT", s.route+"/api/v1/signup/verification", otherToken,
```
Comment (alexeykazakov) [inline test/e2e/registration_service_test.go:518]: Let's also check the error message returned by this endpoint.

**Review by alexeykazakov — COMMENTED** [review]
Looks good overall. But let's check the error message too.

File: test/e2e/registration_service_test.go (hunk 1 above)
Comment (sbryzak) [reply test/e2e/registration_service_test.go]: Done

**Review by alexeykazakov — COMMENTED** [review]
The test looks good!
But let's also test the case when there is a deactivated signup with the same phone and it's ignored.

File: test/e2e/registration_service_test.go [hunk 2]
```diff
@@ -495,6 +495,65 @@ func (s *registrationServiceTestSuite) TestPhoneVerification() {
...
+	// Confirm there is no verification code annotation value
+	require.Empty(s.T(), otherUserSignup.Annotations[v1alpha1.UserSignupVerificationCodeAnnotationKey])
+
+	// Retrieve the current UserSignup
+	err = s.hostAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: s.hostAwait.Namespace, Name: userSignup.Name}, userSignup)
+	require.NoError(s.T(), err)
+
+	// Now mark the original UserSignup as deactivated
+	userSignup.Spec.Deactivated = true
+	userSignup.Labels[v1alpha1.UserSignupStateLabelKey] = v1alpha1.UserSignupStateLabelValueDeactivated
```
Comment (alexeykazakov) [inline test/e2e/registration_service_test.go:541]: No need to set this label. It will be set by the controller.

**Review by alexeykazakov — APPROVED** [review]
Looks good. Just one minor comment.

**Pull request merged**